utils/_context/containers.py

```diff
@@ -132,7 +136,39 @@ def stop_previous_container(self):
             logger.debug(f"Kill old container {self.container_name}")
             old_container.remove(force=True)
 
-    def start(self) -> Container:
+    def start(self) -> Thread:
+        return self._start_async([])
+
+    def _start_async(self, seen: list):
+        """ Start the container and its dependencies in a thread with circular dependency detection """
+        if self in seen:
+            dependencies = " -> ".join([s.name for s in seen])
+            pytest.exit(f"Circular dependency detected between containers: {dependencies}", 1)
+
+        seen.append(self)
+
+        with self._starting_lock:
+            if self._starting_thread is None:
+                self._starting_thread = Thread(target=self._start_with_dependencies, args=(seen,))
+                self._starting_thread.start()
+
+            return self._starting_thread
```

Comment:
Even if it literally the same execution flow, it's cleaner to leave the function outside the context manager 

```suggestion
        return self._starting_thread
```